server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -380,9 +380,17 @@ public void onNewInfo(ClusterInfo info) {
         if ((state.getBlocks().hasGlobalBlockWithId(Metadata.CLUSTER_CREATE_INDEX_BLOCK.id()) == false)
             && nodes.size() > 0
             && nodesOverHighThreshold.size() == nodes.size()) {
+            logger.warn(
+                "Enabling index create block on cluster as all nodes are breaching high disk watermark. "
+                    + "Number of nodes above high watermark: {}. Total numbers of node: {}.",
+                nodesOverHighThreshold.size(),
+                nodes.size()
+            );
             setIndexCreateBlock(listener, true);
         } else if (state.getBlocks().hasGlobalBlockWithId(Metadata.CLUSTER_CREATE_INDEX_BLOCK.id())
-            && diskThresholdSettings.isCreateIndexBlockAutoReleaseEnabled()) {
+            && diskThresholdSettings.isCreateIndexBlockAutoReleaseEnabled()
+            && nodesOverHighThreshold.size() != nodes.size()) {
```

Comment:
Is considering a B/G relevant here? Does that change how this block guard should be interpreted? 

What I am trying to evaluate is if `nodesOverHighThreshold.size() < nodes.size()` is a better check here instead of `nodesOverHighThreshold.size() != nodes.size()` OR if they would end up being equivalent for our use case.

Reply:
@BhumikaSaini-Amazon I agree `nodesOverHighThreshold.size() < nodes.size()` will be better check here.
@psychbot It does not matter, Right? In that case, we will check if active total number of nodes above high watermark is less than active total number of nodes. Once the affected nodes starts responding, `DiskThresholdMonitor` will start considering those nodes as well for evaluation and will add or remove nodes.

Reply:
@BhumikaSaini-Amazon I dont think it is possible. Node will always be accounted in `nodesOverHighThreshold` if it is accounted in `nodes` and node is breaching high watermark. Since we are [iterating over all the nodes](https://github.com/opensearch-project/OpenSearch/blob/cf02b9677fde73056e071c6aaae2dd67b227ffe3/server/src/main/java/org/opensearch/cluster/routing/allocation/DiskThresholdMonitor.java#L216) and evaluating if it is breaching any watermarks.